test/test_matmul_cuda.py

```diff
@@ -445,6 +454,47 @@ def data_to_mx_scale(x, block_size):
     return scale_e8m0_biased.reshape(orig_shape[0], -1)
 
 
+def data_to_nvfp4_scale(x, block_size):
+    orig_shape = x.shape
+    x = x.reshape(-1, block_size)
+    max_abs = torch.amax(torch.abs(x), 1) + 1e-12
+
+    # x_orig_max / scale = x_in_fp4_domain_max
+    # x_orig_max / x_in_fp4_domain_max = scale
+    scale = max_abs / FP4_MAX_VAL
+
+    # for the purposes of this function, just clamp to representable range of
+    # `torch.float8_e4m3fn`. In real code, we would expect the modeling code to
+    # handle this before the input data hits this function.
+    scale = scale.clamp(max=F8E4M3_MAX_VAL)
+
+    # cast to target dtype
+    scale = scale.to(torch.float8_e4m3fn)
+    scale = scale.reshape(orig_shape[0], -1)
+    return scale
+
+
+def down_size(size):
```

Comment:
nit: this is a little confusing, does the format support odd number of fp4 elements and if it did wouldn't it be 

data .... (0123,4567). where 4567 are meangingless but the [0-3] least significant bits hold the value?

Reply:
to clarify, are you asking about `torch.float4_e2m1fn_x2`, or `down_size`?

Currently I think neither of them should support odd number of fp4 elements:
1. for `torch.float4_e2m1fn_x2`, it's up to the user to decide what to do with odd number of elements, as this format is packed.
2. for `down_size` we do have an assert on L478 for not supporting odd number of elements

Reply:
I am curious if the fp4x2 is is specified if it has an odd number of elements, specified meaning that the cublas kernel supports this case

Reply:
Just tried calling the cuBLAS kernel with an odd value of `k`, I see

```bash
CUDA error: CUBLAS_STATUS_NOT_SUPPORTED when calling `cublasLtMatmulAlgoGetHeuristic(...
```